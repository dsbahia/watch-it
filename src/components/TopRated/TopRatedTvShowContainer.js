import React, { useEffect, useState } from "react";
import { toast } from "react-hot-toast";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCircleChevronDown } from "@fortawesome/free-solid-svg-icons";
import TopRatedTvShow from "./TopRatedMovies";
import api from "../../api/api";
import "../../styles/topratedmoviescontainer.css";

function TopRatedTvShowContainer() {
  const [moviesData, setMoviesData] = useState([]);
  const [maxResults, setMaxResults] = useState(9);

  useEffect(() => {
    async function fetchTopRatedTvShow() {
      try {
        const data = await api.topRatedTvShows();
        setMoviesData(data.results);
      } catch (error) {
        const errorMsg = "An error occurred. Please try again later.";
        toast.error(errorMsg, {
          duration: 5000,
        });
      }
    }
    fetchTopRatedTvShow();
  }, []);

  const handleShowMore = () => {
    setMaxResults(maxResults + 6);
  };
  return (
    <div className="top-rated-container">
      <div className="top-rated-movie-title">Top Rated Tv Shows</div>
      <div className="top-rated-results">
        {moviesData.slice(0, maxResults).map((data) => (
          <div key={data.id} className="top-rated-item">
            <TopRatedTvShow
              title={data.original_name}
              posterpath={data.poster_path}
              movieId={data.id}
            />
          </div>
        ))}
      </div>{" "}
      {maxResults < moviesData.length && (
        <button
          type="button"
          className="show-more-button"
          onClick={handleShowMore}
        >
          <FontAwesomeIcon
            icon={faCircleChevronDown}
            size="2xl"
            style={{ color: "#393E46" }}
          />
        </button>
      )}
    </div>
  );
}

export default TopRatedTvShowContainer;
