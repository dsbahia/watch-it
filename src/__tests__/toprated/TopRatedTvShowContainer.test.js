import React from "react";
import { render, screen } from "@testing-library/react";
import TopRatedTvShowContainer from "../../components/TopRated/TopRatedTvShowContainer";

describe("TopRatedTvShowContainer", () => {
  it("should render without errors", () => {
    render(<TopRatedTvShowContainer />);
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<TopRatedTvShowContainer />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays a list of top rated movies", async () => {
    render(<TopRatedTvShowContainer />);

    const movieList = await screen.findByTestId("tv-list");

    expect(movieList).toBeInTheDocument();
  });
});