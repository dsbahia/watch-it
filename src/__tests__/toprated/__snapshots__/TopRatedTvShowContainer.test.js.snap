// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopRatedTvShowContainer matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="top-rated-container"
    data-testid="tv-list"
  >
    <div
      class="top-rated-movie-title"
    >
      Top Rated Tv Shows
    </div>
    <div
      class="top-rated-results"
    />
     
  </div>
</DocumentFragment>
`;
